aten/src/ATen/native/Convolution.cpp

```diff
@@ -146,7 +148,31 @@ auto ConvParams::use_cpu_depthwise3x3_winograd(
 #endif
 }
 
-auto ConvParams::use_cudnn(const at::Tensor& input) const -> bool {
+auto ConvParams::needs_64bit_indexing_no_split(const at::Tensor& input, const at::Tensor& weight) const -> bool {
+  constexpr int64_t int_max = std::numeric_limits<int>::max();
+  int64_t numel_input = input.numel();
+  // empty input
+  if (numel_input == 0) {
```

Comment:
Do we need a separate check for an empty input here? At worst it will produce a negative output size and send a convolution to cudnn that will error out, as it's supposed to.

Reply:
The worst case is `n = 0` and `numel_input / n > int_max` triggers a divide by zero error. So I just filter out this case and defer the handle of empty tensor to the rest of the code